tests/framework/validators/CompareValidatorTest.php

```
@@ -37,364 +43,836 @@ protected function tearDown(): void
+            [
+                'operator' => '==',
+                'type' => 'string',
+                'compareValue' => 'test_value',
+                'skipOnEmpty' => 1,
+                'message' => 'attrA must be equal to "test_value".',
+            ],
+            $validator->getClientOptions($model, 'attrA'),
+            "'getClientOptions()' should return correct options array.",
+        );
+
+        $validator->validate('someIncorrectValue', $errorMessage);
```

PHPMD (2.15.0)

[warning] Avoid using undefined variables such as '$errorMessage' which will lead to PHP notices. (undefined)

(UndefinedVariable)